charts/opentelemetry-operator/UPGRADING.md

```diff
@@ -1,5 +1,10 @@
 # Upgrade guidelines
 
+## <0.42.1 to 0.42.1
+
+A type of flag `autoGenerateCert` has been changed, now it is an object with two attributes `enabled` and `recreate`.
+If you previously set `autoGenerateCert` to `true` or `false` you have to set `autoGenerateCert.enabled` accordingly.
```

Comment:
```suggestion
If you previously set `autoGenerateCert=true`, no changes need. Otherwise, set `autoGenerateCert.enabled=false`.
In addition, you can use `autoGenerateCert.recreate` to decide whether to regenerate the cert or not.
```

Reply:
true is a default value, but if someone explicitly set it, upgrading chart will fail too...

```
helm upgrade my-test open-telemetry/opentelemetry-operator --install --set admissionWebhooks.autoGenerateCert=true

helm upgrade  my-test . --install --reuse-values

- admissionWebhooks.autoGenerateCert: Invalid type. Expected: object, given: boolean
```

setting the same value as a default value is not common so we can ignore it, right?

Reply:
Setting the default value is not common, but some users do choose to include the whole values.yaml file in their config. I like your original text that references both values.